syntax = "proto3";

message SportRoot_s {
    sport_data_s data = 1;
}

message sport_data_s {
    sport_data_sport_s data_sport = 1;
}

message sport_data_sport_s {
    sport_sport_s sport = 1;
    int32 number = 3;
    repeated sport_category_events_s category = 4;
}

message sport_sport_s {
    string sport_id = 1;
    string sport_name = 2;
}

message sport_category_events_s {
    sport_events_info_s events_info = 1;
}

message sport_events_info_s {
    string events_xx = 1;
    string events_category = 2;
    string events_x = 3;
    repeated sport_event_s events = 5;
}

message sport_event_s {
    int64 event_id = 1;
    string event_name = 2;
    sport_event_time_s event_time = 4;
    string number_5 = 5;
    sport_bet_details_s bet_details = 6;
    string event_xx = 7;
    int32 number_10 = 10;
    int32 number_12 = 12;
}

message sport_event_time_s {
    int64 datetime = 1;
}

message sport_bet_details_s {
    int32 bet_category = 1;
    string bet_category_name = 2;
    sport_bet_s bet = 3;
}

message sport_bet_s {
    string bets_id = 1;
    string bets_name = 2;
    sport_bets_info_s bets_info = 6;
    int32 number_11 = 11;
    int32 number_12 = 12;
}

message sport_bets_info_s {
    int64 bet_id = 1;
    string bet_name = 2;
    int32 bet_number = 3;
    repeated sport_odd_s odds = 4;
}

message sport_odd_s {
    string odd_id = 1;
    string odd_name = 2;
    float odd_number = 3;
}